File: Nodejs-server/src/routes/songRoutes.js
const express = require("express");
const {
  createSong,
  getAllSongs,
  getSongById,
  updateSong,
  deleteSong,
  getSongByName,
} = require("../controllers/songController");
const router = express.Router();

router.post("/", createSong);
router.get("/", getAllSongs);
router.get("/:id", getSongById);
router.get("/byName/:name", getSongByName);
router.put("/:id", updateSong);
router.delete("/:id", deleteSong);

module.exports = router;

File: Python-server/src/routes/songRoutes.py
from flask import Blueprint, request, jsonify
import controllers.songController as songController

songController_bp = Blueprint('songController_bp', __name__, url_prefix = '/api/songs')

@songController_bp.route('/', methods=['POST'])
def createSong():
    return songController.createSong()

@songController_bp.route('/<int:id>', methods=['PUT'])
def updateSong(id):
    return songController.updateSong(id)

@songController_bp.route('/', methods=['GET'])
def getAllSongs():
    return songController.getAllSongs()

@songController_bp.route('/<int:id>', methods=['GET'])
def getSongById(id):
    return songController.getSongById(id)

@songController_bp.route('/<int:id>', methods=['DELETE'])
def deleteSong(id):
    return songController.deleteSong(id)